### PR TITLE
Add softPurgeKey method

### PR DESCRIPTION
**Pull request by sisidovski**
Hi

I added [soft purge](https://docs.fastly.com/api/purge#soft_purge_2e4d29085640127739f8467f27a5b549) method to this library. Original repo have already supported soft-purge, but I want to use ft forked one because this API returns promise, has a feature to manipulate VCL, etc.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -130,6 +131,11 @@ Fastly.prototype.purgeKey = function (service, key) {
     return this.request('POST', url);
 };
 
+Fastly.prototype.softPurgeKey = function (service, key) {
+    var url = '/service/' + encodeURIComponent(service) + '/purge/' + key;
```
Comment (lucas42) [inline lib/index.js]: Please can we encode the key too?

File: lib/index.js [hunk 2]
```diff
@@ -81,7 +82,7 @@ Fastly.prototype.request = function (method, url, params) {
     debug('Request: ' + method + ', ' + url);
 
     // Construct headers
-    var headers = { 'x-fastly-key': self.apikey };
+    headers = Object.assign({}, { 'x-fastly-key': self.apikey }, headers || {});
```
Comment (lucas42) [inline lib/index.js]: I don't think there's any need for the first parameter here.

File: lib/index.js (hunk 1 above)
Comment (sisidovski) [reply lib/index.js]: Sure

File: lib/index.js (hunk 2 above)
Comment (sisidovski) [reply lib/index.js]: Right. That isn't necessary.

**Pull request merged**